util/site-dashboard/ot-nightly-results.js

```diff
@@ -71,14 +91,19 @@ async function fetch_results(block_report_urls) {
     .map((report_result) => report_result.value);
 }
 
-async function fetch_block_results(block_name) {
-  let block_report_urls = all_report_urls.find((block_info) => block_info[0] == block_name);
 
-  return fetch_results(block_report_urls[1]);
-}
-
-async function fetch_all_results() {
-  block_report_urls = all_report_urls.flatMap((url_info) => url_info[1]);
-
-  return fetch_results(block_report_urls);
+/**
+ * This function is used for the block-dashboard on the homepage of each HWIP block.
+ *
+ * @param {string} block_name - The name of the block to generate the dashboard for.
+ * @returns {Array[BlockLevelResults]} A array with the processed block report(s) for all variants.
+ */
+async function fetch_block_results(block_name) {
+    // Filter-out all the blocks that don't match the 'block_name' argument
+    const block_report_urls = Object.fromEntries(
+        Object.entries(block_level_urls).filter(
+            ([key, val]) => key.includes(block_name)
+        )
+    );
```

Comment:
Is the idea here that `block_name = "rstmgr"` matches the URLS for both `rstmgr` and `rstmgr_cnsty_chk` (for example)?

Reply:
yes that's the idea, it seems to work for our naming scheme for now. 
It should also work for prim but for some reason we don't render the dashboards on the [prim page](https://opentitan.org/book/hw/ip/prim/index.html).